src/traces/funnel/calc.js

```diff
@@ -0,0 +1,89 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Axes = require('../../plots/cartesian/axes');
+var arraysToCalcdata = require('./arrays_to_calcdata');
+var calcSelection = require('../scatter/calc_selection');
+var BADNUM = require('../../constants/numerical').BADNUM;
+
+module.exports = function calc(gd, trace) {
+    var xa = Axes.getFromId(gd, trace.xaxis || 'x');
+    var ya = Axes.getFromId(gd, trace.yaxis || 'y');
+    var size, pos, i, cdi;
+
+    if(trace.orientation === 'h') {
+        size = xa.makeCalcdata(trace, 'x');
+        pos = ya.makeCalcdata(trace, 'y');
+    } else {
+        size = ya.makeCalcdata(trace, 'y');
+        pos = xa.makeCalcdata(trace, 'x');
+    }
+
+    // create the "calculated data" to plot
+    var serieslen = Math.min(pos.length, size.length);
+    var cd = new Array(serieslen);
+
+    trace._base = [];
```

Comment:
Maybe add comment above this line to describe how `_base` is different than `base` for funnel traces?

Reply:
Good call. Done in 4fb5a2b166d7cfeda81a1783c71b1d550c0760ca.

Reply:
Thanks!